### DISCUSSION TITLE
Contribution Topics

### DISCUSSION THREAD

**Discussion by michaelstaib** [ChilliCream/hotchocolate#4869] [Ideas]
# Stitching
- Schema Transform / Merge
- MetaDB
- Execution Plan Integration
- https://github.com/ChilliCream/hotchocolate/issues/3586
- OData Adapter for Stitching

# Performance
- Execution Engine Optimization
- Validation Performance (https://github.com/ChilliCream/hotchocolate/issues/4040)
- Parser Memory Improvements
- https://github.com/ChilliCream/hotchocolate/issues/2962

# Quality Topics
- Missing Tests
- Fix Warnings
- Code Style Fixes
- Code Documentation
- XML documentation of most used APIs (Descriptors for example)
- Documenting one of the topics mentioned [here](https://github.com/ChilliCream/hotchocolate/issues/4209)

# StrawberryShake
- Defer / Stream
- File Upload (https://github.com/ChilliCream/hotchocolate/issues/3312)
- https://github.com/ChilliCream/hotchocolate/issues/4771
- https://github.com/ChilliCream/hotchocolate/issues/4759
- https://github.com/ChilliCream/hotchocolate/issues/4754
- https://github.com/ChilliCream/hotchocolate/issues/4596
- https://github.com/ChilliCream/hotchocolate/issues/3985

# Type System
- Improve XML to GraphQL Description (https://github.com/ChilliCream/hotchocolate/issues/3180,  https://github.com/ChilliCream/hotchocolate/issues/3730, https://github.com/ChilliCream/hotchocolate/issues/4070)
- schema-first

# Spec Changes
- Introducing Merge Rules for Input #285

# Data

- Bring back operation toggles on fields (`descriptor.Field(x => x.Foo).AllowEqual().AllowNotEqual()`) (https://github.com/ChilliCream/hotchocolate/issues/3434)
- Introducing a source generator for Expressions. This will be used in a lot of different places, like filtering and sorting on computed fields and later also on projections.             This will also be used to prototype an easier way of adding custom filtering operation for the Expression Provider (that is used in e.g. EF-Core). Reference: [Expressionify](https://github.com/ClaveConsulting/Expressionify) and [EntityFrameworkCore.Projectables](https://github.com/koenbeuk/EntityFrameworkCore.Projectables/tree/401b7d804387489dd43331cbfc0ef9250c470c97)
- #4018
- https://github.com/ChilliCream/hotchocolate/issues/3557
# Spatial

- Sorting on spatial types (Sort by distance) (https://github.com/ChilliCream/hotchocolate/issues/1461)

# Documentation
- Subscription Authentication (https://github.com/ChilliCream/hotchocolate/issues/1732)
- https://github.com/ChilliCream/hotchocolate/issues/3929

# Database Specific Topics
## MonogDb
- https://github.com/ChilliCream/hotchocolate/issues/3898
- https://github.com/ChilliCream/hotchocolate/issues/3881

## Postgres
- https://github.com/ChilliCream/hotchocolate/issues/4429

## Elastic Search
- Join the discussion here :https://github.com/ChilliCream/hotchocolate/discussions/4883

**Comment by matt-psaltis** [comment]
Hey all coming at this from my current pain points mostly while I get across the larger vision :) 
* Super graph partial loading / streaming / partitioning and sharding. All about being a good citizen in the cloud where memory isn't as cheap as the brochure says it is. :) 
* flattened stitching everything in root. "just works" consistent collision renaming. Current collision algorithm is indeterminate. It's based on order of schema publishing from downstream.
* ~~nested stitching more like gql namespacing I believe this is still controversial?~~
* existing renaming process doesn't rename enums when names collide.
* bcp: documentation mode take away all the dev centric components pure, unadulterated schema documentation
* ss: simplified multi clients per project
* ss / hc: wrapping async as sync (request/response)

**Reply by michaelstaib** [reply]
> nested stitching more like gql namespacing I believe this is still controversial?

Yes, since it would entail something like nested mutations.

In any case, I think this is a bad idea since the idea with GraphQL is to have one consistent graph. With this we would essentially create a simple HTTP gateway that just does top-level routing.

**Comment by matt-psaltis** [comment]
Do we currently have any details / designs for the metadb?

**Reply by michaelstaib** [reply]
Not yet... The idea here is to have an in-memory read-only store that holds all the information for the gateway. Essentially we will read the supergraph and create from that the metadb. We do not need to have all type-information since we have an ISchema instance, but we need all additional information.

Shall we do a planning meeting on stitching to sketch things out or shall we do it async in a separate discussion?

**Reply by michaelstaib** [reply]
The topics are just starting points, from there we need to flash them out. I already have a prototype implementation of some things.

This is the very early prototyping:
https://github.com/ChilliCream/hotchocolate/tree/Gateway/src/HotChocolate/Stitching.Next

However, we will not build on that branch, it was more a proof of concept of certain concepts.
